python/tk_houdini/utils/parm_template_wrappers.py

```diff
@@ -0,0 +1,223 @@
+from __future__ import print_function
+
+
+def wrap_node_parameter_group(node):
+    """
+    Helper function to convert a node's parameter group into
+    a ParmGroup instance.
+
+    :param node: A :class:`hou.Node` instance.
+
+    :returns: A :class:`ParmGroup` instance.
+    """
+    template_group = node.parmTemplateGroup()
+    parameter_group = ParmGroup(template_group)
+    return parameter_group
+
+
+class Parm(object):
+    """
+    Wrapper class for houdini parameter template instances.
+    """
+
+    @classmethod
+    def build_parm(cls, template):
+        """
+        Method to build a Parm object from a given template.
+
+        :param template: A :class:`hou.ParmTemplate` instance.
+
+        :returns: A :class:`Parm` or :class:`ParmFolder` instance.
+        """
+        import hou
+
+        if template.type() == hou.parmTemplateType.Folder:
+            return ParmFolder(template)
+        return Parm(template)
+
+    def __init__(self, template):
+        """
+        Initialise the class.
+
+        :param template: A :class:`hou.ParmTemplate` instance.
+        """
+        self.template = template
+
+    @property
+    def name(self):
+        """
+        Get the template name.
+
+        :rtype: str
+        """
+        return self.template.name()
+
+    def build(self):
+        """
+        Get the template relating to this Parm object.
+
+        :rtype: :class:`hou.ParmTemplate`
+        """
+        return self.template
+
+    def dumps(self, indent=0):
+        """
+        Dump a string representation of this object for debugging.
+
+        :param int indent: Size of indenation.
+        """
+        print("\t" * indent, self.template)
+
+
+class ParmFolder(Parm):
+    """
+    Wrapper class for houdini folder parameter template instances.
+    """
+
+    def __init__(self, template):
+        """
+        Initialise the class.
+
+        :param template: A :class:`hou.ParmTemplate` instance.
+        """
+        super(ParmFolder, self).__init__(template)
+        self.__children = [Parm.build_parm(child) for child in template.parmTemplates()]
+        self.__child_names = [child.name for child in self.__children]
+
+    def index_of_template(self, template_name):
+        """
+        Get the index of the child template from the list of child templates.
+
+        :param str template_name: The name of the template to find.
+        :rtype: int
+        """
+        return self.__child_names.index(template_name)
+
+    def extend_templates(self, templates):
+        """
+        Add multiple templates to the child templates list..
+
+        :param list(hou.ParmTemplate) templates: The list of template
+            instances to add.
+        """
+        template_names = [template.name() for template in templates]
+        child_templates = [Parm.build_parm(template) for template in templates]
```

Comment:
I think it might be faster if you switched these to generators
```suggestion
        template_names = (template.name() for template in templates)
        child_templates = (Parm.build_parm(template) for template in templates)
```